lightning/src/offers/static_invoice.rs

```diff
@@ -102,8 +102,8 @@ pub struct StaticInvoiceBuilder<'a> {
 impl<'a> StaticInvoiceBuilder<'a> {
 	/// Initialize a [`StaticInvoiceBuilder`] from the given [`Offer`].
 	///
-	/// Unless [`StaticInvoiceBuilder::relative_expiry`] is set, the invoice will expire 24 hours
-	/// after `created_at`.
+	/// The invoice's expiration will default to [`DEFAULT_RELATIVE_EXPIRY`] after `created_at` unless
+	/// overridden by [`StaticInvoiceBuilder::relative_expiry`].
```

Comment:
IIUC, this default is not defined by the spec (which uses 2 hours for any invoice), and thus if not set `StaticInvoice:: relative_expiry` would give 24 hours whereas another implementation would assume it is 2 hours.

Reply:
Good point, thanks. I'll need to update the spec for that (added to the tracking issue for now). Currently `relative_expiry()` gives 2 weeks if not set, though we may want to revisit that.

Reply:
Do we still need to explicitly set it in the `ChannelManager` utility then as was added in the recent push? I guess it doesn't matter much since size isn't a constraint, but just wondering if a test caught this in a follow-up?

Reply:
Yeah, caught this in testing: https://github.com/lightningdevkit/rust-lightning/pull/3517/commits/7d239c4f2507bc4ec81ddede2a1e81d51472a397. Fair that we could avoid setting it if it's `None` or the default, though, let me know if you'd prefer that. Since size isn't a constraint as you mention, it simplifies the code a bit to always set it so I'm neutral.

Reply:
Might as well have it be explicit.